CPP_STYLE_GUIDE.md

```diff
@@ -166,18 +167,34 @@ class FancyContainer {
  ...
 }
 ```
-## Memory Management
 
-### Memory allocation
+## Semantics
+
+The Node.js C & C++ code strives to be consistent in it's use of language
+features and idioms, as well as have some specific guidelines for the use of
+runtime features.
+
+Coding guidelines are based of following guides (highest priority first):
+1. This guide
+2. The [Google C++ Style Guide][](GCSG)
+3. The ISO [C++ Core Guidelines][](CPPCG)
+
+In general code should follow the CPPCG, unless overridden by the GCSG or this
+document. ATM these guidelines are checked manually by reviewers, but the goal
+is to automate this check via tools such as `clang-tidy`.
```

Comment:
I wouldn't mention `clang-tidy` as a goal here since the last time I attempted to use it on our code base it took really long to complete linting...certainly much longer than cpplint and cpplint is already kind of slow.

Reply:
Ack.
Recent versions seems to work much much better. MSVS has it built in, and it runs in reasonable time (for just `/src/` and with a limited ruleset). It does spit out a huge amount of recommendations.